File: libxrme/remotecontrolinterface.cpp
#include "remotecontrolhandler.h"
#include "remotecontrolinterface.h"

namespace xrme {

struct RemoteControlInterface::Private {
  Private()
    : handler_(NULL) {}

  RemoteControlHandler* handler_;
};


RemoteControlInterface::RemoteControlInterface()
    : d(new Private) {
}

RemoteControlInterface::~RemoteControlInterface() {
}

void RemoteControlInterface::PlayPause(const QString& peer_jid_resource) {
  if (d->handler_) {
    d->handler_->PlayPause(peer_jid_resource);
  }
}

void RemoteControlInterface::Stop(const QString& peer_jid_resource) {
  if (d->handler_) {
    d->handler_->Stop(peer_jid_resource);
  }
}

void RemoteControlInterface::Next(const QString& peer_jid_resource) {
  if (d->handler_) {
    d->handler_->Next(peer_jid_resource);
  }
}

void RemoteControlInterface::Previous(const QString& peer_jid_resource) {
  if (d->handler_) {
    d->handler_->Previous(peer_jid_resource);
  }
}

void RemoteControlInterface::QueryState(const QString& peer_jid_resource) {
  if (d->handler_) {
    d->handler_->QueryState(peer_jid_resource);
  }
}

void RemoteControlInterface::Attach(RemoteControlHandler* handler) {
  d->handler_ = handler;
}

} // namespace xrme

File: java-android/src/com/purplehatstands/libxrme/RemoteControlInterface.java
package com.purplehatstands.libxrme;

import android.graphics.Bitmap;

public abstract class RemoteControlInterface {
  public abstract void StateChanged(String peer_jid_resource, State state);
  public abstract void AlbumArtChanged(String peer_jid_resource, Bitmap image);
  
  protected void PlayPause(String peer_jid_resource) {
    if (handler_ != null) {
      handler_.PlayPause(peer_jid_resource);
    }
  }
  
  protected void Stop(String peer_jid_resource) {
    if (handler_ != null) {
      handler_.Stop(peer_jid_resource);
    }
  }
  
  protected void Next(String peer_jid_resource) {
    if (handler_ != null) {
      handler_.Next(peer_jid_resource);
    }
  }

  protected void Previous(String peer_jid_resource) {
    if (handler_ != null) {
      handler_.Previous(peer_jid_resource);
    }
  }
  
  protected void QueryState(String peer_jid_resource) {
    if (handler_ != null) {
      handler_.QueryState(peer_jid_resource);
    }
  }
  
  public void Attach(RemoteControlHandler handler) {
    handler_ = handler;
  }
  
  private RemoteControlHandler handler_;
}
